data-processing/entities/definitions/model/load_data.py

```diff
@@ -555,6 +555,80 @@ def load_and_cache_example(
     )
     return dataset
 
+def load_and_cache_example_batch_raw(
```

Comment:
This returns the raw data as well (along with the tensor dataset). Raw data is needed in the filtering heuristics stage. I could have edited the old function to do this but it would break at places and would require a change to the old code as well.